webclient/src/websocket/services/WebSocketService.ts

```diff
@@ -60,7 +61,10 @@ export class WebSocketService {
     const socket = new WebSocket(url);
     socket.binaryType = 'arraybuffer';
 
+    const connectionTimer = setTimeout(() => socket.close(), this.keepalive);
```

Comment:
Is `socket.error()` a more appropriate choice here?  I can't quite tell, but this implementation suggests to me that if the code is ever called it's unexpected and represents an unexpected error.

Reply:
From testing this out, calling `socket.close` before a connection is made results in an error being thrown anyway. This only gets called if the connection isnt made within the time limit. This was to prevent the 30+ second default timeout that webSockets limit themselves to, we basically just cancel the request ourself with `socket.close`.

Reply:
Fair enough.  Thanks for the enlightenment.